### PR TITLE
(parcel-bundler) Add serve method to parcel-bundler types

### PR DESCRIPTION
**Pull request by hasparus**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 

* https://en.parceljs.org/api.html
* https://github.com/parcel-bundler/parcel/blob/master/packages/core/parcel-bundler/src/Bundler.js#L787

- [ ] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. -- Already exists.

### CODE REVIEW

File: types/parcel-bundler/index.d.ts [hunk 1]
```diff
@@ -173,6 +179,12 @@ declare class ParcelBundler {
     addPackager(type: string, packager: string): void;
 
     bundle(): Promise<ParcelBundler.ParcelBundle>;
+
+    serve(
+        port?: number,
+        https?: boolean | { cert: string; key: string },
+        host?: string
+    ): http.Server | https.Server;
```
Comment (pinage404) [inline types/parcel-bundler/index.d.ts]: [the method is asynchronous](https://github.com/parcel-bundler/parcel/blob/6cbf2a65cdcd66af135ff9eca2a3799bde9f8a5b/packages/core/parcel-bundler/src/Bundler.js#L787)

```js
async serve(port = 1234, https = false, host) {
```

so, i think, it should be like that

```ts
Promise<http.Server | https.Server>;
```

File: types/parcel-bundler/index.d.ts [hunk 2]
```diff
@@ -173,6 +179,12 @@ declare class ParcelBundler {
     addPackager(type: string, packager: string): void;
 
     bundle(): Promise<ParcelBundler.ParcelBundle>;
+
```
Comment (pinage404) [inline types/parcel-bundler/index.d.ts]: Maybe you can add a description

e.g. something like this (or better)

```ts
/**
 * Starts a simple http (or https) server.
 * 
 * @param `https` `{ cert, key }` pointing to the location of key and cert file or `true` to generate a key
 */
```

i copied and adapted from [official web site](https://en.parceljs.org/api.html)

File: types/parcel-bundler/index.d.ts (hunk 1 above)
Comment (hasparus) [reply types/parcel-bundler/index.d.ts]: Damn, you're totally right @pinage404. Thanks!

File: types/parcel-bundler/index.d.ts [hunk 3]
```diff
@@ -173,6 +184,23 @@ declare class ParcelBundler {
...
+    serve(
+        port: number,
+        https: true | HttpsInfo,
+        host?: string
+    ): Promise<https.Server>;
+    serve(
+        port?: number,
+        https?: boolean | HttpsInfo,
+        host?: string
+    ): Promise<http.Server | https.Server>;
```
Comment (PranavSenthilnathan) [inline types/parcel-bundler/index.d.ts]: The 2nd overload doesn't have optionals for the first 2 parameters but I think the JS library allows them to be optional. If this is in fact the case, this function can be written as:

```ts
serve<T extends boolean | HttpsInfo>(sort?: number, https?: T, host?: string): Promise<T extends false ? http.Server : https.Server>
```

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

**Pull request closed without merging**